cmd/end2endtest/encrypted.go

```diff
@@ -0,0 +1,381 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	vapi "go.vocdoni.io/dvote/api"
+	"go.vocdoni.io/dvote/apiclient"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/types"
+	"go.vocdoni.io/dvote/util"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+func init() {
+	ops["encryptedelection"] = operation{
+		test:        &E2EEncryptedElection{},
+		description: "Publishes a census and a non-anonymous, secret-until-the-end election, emits N votes and verifies the results",
+		example:     os.Args[0] + " --operation=encryptedelection --votes=1000",
+	}
+}
+
+var _ VochainTest = (*E2EEncryptedElection)(nil)
+
+type E2EEncryptedElection struct {
+	api    *apiclient.HTTPclient
+	config *config
+}
+
+func (t *E2EEncryptedElection) Setup(api *apiclient.HTTPclient, config *config) error {
+	t.api = api
+	t.config = config
+	return nil
+}
+
+func (t *E2EEncryptedElection) Teardown() error {
+	// nothing to do here
+	return nil
+}
+
+func (t *E2EEncryptedElection) Run() (duration time.Duration, err error) {
+	start := time.Now()
+
+	c := t.config
+	api := t.api
+
+	// Set the account in the API client, so we can sign transactions
+	if err := api.SetAccount(hex.EncodeToString(c.accountKeys[0].PrivateKey())); err != nil {
+		log.Fatal(err)
+	}
+
+	// If the account does not exist, create a new one
+	// TODO: check if the account balance is low and use the faucet
+	acc, err := api.Account("")
+	if err != nil {
+		log.Infof("getting faucet package")
+		faucetPkg, err := getFaucetPackage(c, api.MyAddress().Hex())
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Create the organization account and bootstraping with the faucet package
+		log.Infof("creating Vocdoni account %s", api.MyAddress().Hex())
+		log.Debugf("faucetPackage is %x", faucetPkg)
+		hash, err := api.AccountBootstrap(faucetPkg, &vapi.AccountMetadata{
+			Name:        map[string]string{"default": "test account " + api.MyAddress().Hex()},
+			Description: map[string]string{"default": "test description"},
+			Version:     "1.0",
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*40)
+		defer cancel()
+		if _, err := api.WaitUntilTxIsMined(ctx, hash); err != nil {
+			log.Fatalf("gave up waiting for tx %x to be mined: %s", hash, err)
+		}
+
+		acc, err = api.Account("")
+		if err != nil {
+			log.Fatal(err)
+		}
+		if c.faucet != "" && acc.Balance == 0 {
+			log.Fatal("account balance is 0")
+		}
+	}
+
+	log.Infof("account %s balance is %d", api.MyAddress().Hex(), acc.Balance)
+
+	// Create a new census
+	censusID, err := api.NewCensus("weighted")
```

Comment:
I think we can use `vapi.CensusTypeWeighted`

Reply:
good one, thanks!